### PR TITLE
Locale separator

### PR DESCRIPTION
**Pull request by aprocom**
As result of conversation at https://github.com/rubenv/angular-gettext/issues/288
Changes to support locales like "en-US" with separator "-".
Separator is configurable.

### CODE REVIEW

File: src/catalog.js [hunk 1]
```diff
@@ -122,6 +122,14 @@ angular.module('gettext').factory('gettextCatalog', function (gettextPlurals, ge
         currentLanguage: 'en',
         /**
          * @ngdoc property
+         * @name gettextCatalog#localeSeparator
+         * @public
+         * @type {String}
+         * @description Language code locale separator. As example "en-US".
+         */
+        localeSeparator: '-',
```
Comment (rubenv) [inline src/catalog.js]: Adding this will cause the default to be `-`, which would be a compatibility break, since we currently have `_`.

**Comment by rubenv** [comment]
Apart from the comment above, I don't think this should be a configuration option. Let's just do the right thing and support both `-` and `_`.

So this:

```js
langCode.split('_').shift();
```

Would become:

```js
langCode.split(/\-|_/).shift();
```

No other changes needed, apart from a unit test.

**Comment by aprocom** [comment]
@rubenv your solution is well, I revert changes and add patch to support both separators.

**Comment by rubenv** [comment]
Cool, could you add a unit test as well? Just copy the two we have here and add a version that uses -: https://github.com/rubenv/angular-gettext/blob/master/test/unit/fallback_language.js

**Comment by rubenv** [comment]
Correction, the tests for plurals are here: https://github.com/rubenv/angular-gettext/blob/master/test/unit/plurals.js

We'll also need to add a hardcoded mapping here: https://github.com/rubenv/angular-gettext/blob/31a175c4d85e793a694bc5dfeb956fccabd05d76/src/plural.js#L3-L5

```
   var languageCodes = {
        "pt_BR": "pt_BR",
        "pt-BR": "pt_BR",
    };
```

**Comment by aprocom** [comment]
@rubenv please look at last changes

**Comment by rubenv** [comment]
Excellent, merged, thanks a lot!

**Comment by rubenv** [comment]
Released as `v2.3.5`.

**Comment by aprocom** [comment]
@rubenv cool, thanks)

**Pull request merged**